docs/design/fabric-v2+/fpc-registration.puml

```diff
@@ -7,266 +7,198 @@
-  We realized FPC-liflecycle as application transaction EP of ERCC chaincode;
-  should ideally match lifecycle EP which is 'Channel/Application/LifecycleEndorsement' and by default majority of organizations)
-  Note that with the **integrated** approach we have deal with a mix of applications transactions EP and FPC-lifecycle EP.
+note over ERCC1, ERCC2 #FFaa00
+  It is assumed that:
+    -channel has already been created;
+    -ERCC has been installed -- approved by the participating organizations, see lifecycle;
+    -ERCC contains hard-coded the TLCC_MRenclave and the IAS Root Certificate;
+    -ERCC has an Endorsement Policy that requires signatures from a majority of orgs,
+     for security reasons -- namely to avoid unilateral modifications and preserve
+     integrity. As a result, changes to chaincode enclave info stored in ERCC will
+     be dependent on such policy.
 end note
 
-ref over User1, ERCC1
-  Create channel; each peer joins channel; each peer launches TLCC, which will  "imprint" itself with CH_id (content-addressable channel identifier, hash of gensis block)
-end ref
-|||
-
-ref over User1, ERCC1
-  Install and approve ERCC chaincode. ERCC defines TLCC_MR and IAS_PK in chaincode definition.
-  (alternatively, TLCC_MR and IAS_PK could be in channel configuration; crucial is that they are approved by humans (admins) on sufficiently many organizations)
-  ERCC also reads CH_id from ledger or, if that does not work, also embeds it into chaincode
-
-  see fpc-lifecycle-v2.puml
-end ref
-|||
-
-
-group enclave creation and attestation generation
-  
+group enclave creation
   activate Admin
 
-  Admin -> Manager1 ++: create (CC_NAME, CC_VER, CH_name)
-  note right
-    CC_NAME: chaincode name
-    CC_VER: chaincode version
-    CH_NAME: channel name (needed by the ECC_Manager to cc2cc ECC
-  end note
-  note over Manager1 #FFaa00
-    **MST**: we might want to replace (here and elsewhere) CC_VER with the sequence
-    number from chain code definition as versions can be re-used in downgrades yet we
-    might or might not want to re-use enclaves?
-  end note
-
-  Manager1 -> Manager1 : get SPID and IAS credentials
-  note right: from local peer config
+  note over Admin #FFaa00
+    These operations are assumed to be performed only after the chaincode definition is
+    committed on the channel for two reasons:
+    1) chaincode invocation may not be available before the definition is committed,
+    2) enclave registration should be feasible only for committed chaincodes
 
-  Manager1 -> ECC1 ++: create (SPID)
-  note left: this is a cc2cc call
+    PS: in a future tighter integration with fabric we could remove the first constraint
+    and could potentially piggyback the steps before the ercc registration (enclave and
+    evidence generation) in the `approveformyorg` and the registration as part of
+    `commit` without having to change the protocol flows.
+  end note
 
+  create E_Manager order 15
```

Comment:
as mentioned already above ... UML experts may complain as this means that e_manager instance is created here ....